### PR TITLE
Writing_Good_Commits.rst: Fixes to Closes

### PR DESCRIPTION
**Pull request by prateeksingh0001**
This helps to avoid errors when newcomers directly
copy commit messages on solving issues that are
not bugs.

Closes https://github.com/coala/coala/issues/3737

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: docs/Developers/Writing_Good_Commits.rst [hunk 1]
```diff
@@ -115,7 +115,7 @@ Example:
 
 ::
 
-    Fixes https://github.com/coala/coala/issues/269
+    Closes https://github.com/coala/coala/issues/269
```
Comment (dracarys09) [inline docs/Developers/Writing_Good_Commits.rst]: I think this replacement is not required... 
CC @sils @yash-nisar

Comment (adtac) [inline docs/Developers/Writing_Good_Commits.rst]: No it's required - it's splitting up the original example given in line 11-18 into three parts and explaining each part by requoting it.

**Comment by adtac** [comment]
Actually I also think the shortlog is correct - "Update shortlog" is ambiguous while "Fixes to Closes" is exact and precise.

**Comment by adtac** [comment]
"Change Fixes to Closes" would be more apt, but I imagine he ran out of space - perfectly understandable.

File: docs/Developers/Writing_Good_Commits.rst (hunk 1 above)
Comment (dracarys09) [inline docs/Developers/Writing_Good_Commits.rst]: @prateeksingh0001 nm

**Comment by dracarys09** [comment]
@adtac ```Writing_Good_Commits.rst: Change Fixes to Closes``` sounds better... and that's 49 chars... :stuck_out_tongue:

File: docs/Developers/Writing_Good_Commits.rst (hunk 1 above)
Comment (yash-nisar) [inline docs/Developers/Writing_Good_Commits.rst]: The newcomer may explore https://github.com/coala/coala/issues/269 to see that it is a bug. So should we change the link to an issue that should actually be addressed by `Closes` ? @adtac @dracarys09

Comment (adtac) [inline docs/Developers/Writing_Good_Commits.rst]: Sure, maybe even link to this PR's issue! Meta-recursive examples :D

Comment (yash-nisar) [inline docs/Developers/Writing_Good_Commits.rst]: Yeah, so basically this needs to be restructured with an actual issue, that should be addressed by `Closes`, with the required commit shortlog, body and an issue reference or maybe we could take into consideration the `Quick reference` example and split it into shortlog, message and issue reference. @prateeksingh0001

Comment (prateeksingh0001) [inline docs/Developers/Writing_Good_Commits.rst]: Sure, we can consider the Quick Reference example and then split it into shortlog, message body and issue reference.

**Comment by supergr35** [comment]
needs a rebase

**Comment by prateeksingh0001** [comment]
I'm facing problems while squashing my commits, while performing git rebase, the last commit (b9d651bc63da343effb6e3a0407cf896e5b892d2) which is basically a merge of the branch goodcommits into itself is not being shown. How should I solve this?

**Comment by dracarys09** [comment]
you can use ```git rebase -i HEAD~3``` since you are trying to squash 2 commits... and then change "pick" to "s" for all commits except the first one...

**Comment by supergr35** [comment]
You have have the dreaded merge commit. This is probably because you pushed the rebase(update branch from master) button in github. This is not the proper way of doing this and will break many things. Please do an interactive rebase on the commit before the merge commit and remove the merge commit line(see http://stackoverflow.com/questions/17577409/git-remove-merge-commit-from-history)

**Comment by prateeksingh0001** [comment]
@supergr35  @dracarys09  thanks a lot for helping me out of this problem :). Please tell me if anything else needs to be done :) .

**Comment by dracarys09** [comment]
ack 74fc1b5e9f2a818101914750827c9fb403a02db6

**Comment by dracarys09** [comment]
@prateeksingh0001 please address @yash-nisar comment https://github.com/coala/coala/pull/3744#discussion_r101249871

**Comment by dracarys09** [comment]
unack 74fc1b5e9f2a818101914750827c9fb403a02db6

**Comment by prateeksingh0001** [comment]
So, should I use the Quick Reference example or should I link it to this issue itself for addressing @yash-nisar comment #3744 (comment)

**Comment by yash-nisar** [comment]
IMO, you can use the Quick reference commit, split it into components and change `Fixes https://github.com/coala/coala/issues/5861` to `Closes https://github.com/coala/coala/issues/5861` as that is not pointing to a valid issue link. Also, the user can refer to the More Examples section for a deeper insight. Do you agree @prateeksingh0001 @adtac ?

**Comment by prateeksingh0001** [comment]
Got it, seems good to me :)

**Comment by yash-nisar** [comment]
ack 5b2d0c5b52f9ec11c01767be9df3494398019ff8

File: docs/Developers/Writing_Good_Commits.rst [hunk 2]
```diff
@@ -71,7 +71,7 @@ Example:
 
 ::
 
-    setup: Install .coafile via package_data
+    setup.py: Change bears' entrypoint
```
Comment (Udayan12167) [inline docs/Developers/Writing_Good_Commits.rst]: Unrelated change?

File: docs/Developers/Writing_Good_Commits.rst [hunk 3]
```diff
@@ -96,10 +96,10 @@ Example:
...
 ::
 
-    When installing the .coafile to distutils.sysconfig.get_python_lib, we
-    ignore that this is not the installation directory in every case. Thus
-    it is easier, more reliable and platform independent to let distutils
-    install it by itself.
+    This entrypoint ensures that coala discovers
+    the bears correctly.
+    It helps not writing more functions inside
+    ``coalib`` for this.
```
Comment (Udayan12167) [inline docs/Developers/Writing_Good_Commits.rst]: Unrelated change?

Comment (Udayan12167) [inline docs/Developers/Writing_Good_Commits.rst]: Also the lines should be 72 characters long for the commit description. These look shorter

File: docs/Developers/Writing_Good_Commits.rst (hunk 2 above)
Comment (Udayan12167) [inline docs/Developers/Writing_Good_Commits.rst]: I would suggest you make these changes in another commit. Describing why you are making these changes. Make this commit about only a single thing please.

File: docs/Developers/Writing_Good_Commits.rst (hunk 3 above)
Comment (Udayan12167) [inline docs/Developers/Writing_Good_Commits.rst]: Same as above

File: docs/Developers/Writing_Good_Commits.rst [hunk 4]
```diff
@@ -115,7 +115,7 @@ Example:
 
 ::
 
-    Fixes https://github.com/coala/coala/issues/269
+    Closes https://github.com/coala/coala/issues/5861
```
Comment (Udayan12167) [inline docs/Developers/Writing_Good_Commits.rst]: The 5861 issue number change here is again an unrelated one.

**Comment by prateeksingh0001** [comment]
I have made the changes. Please do tell if any other changes are required. @Udayan12167

**Comment by Udayan12167** [comment]
ack 102a909

**Comment by Udayan12167** [comment]
ack 2541ed1

**Comment by Udayan12167** [comment]
@rultor merge

**Comment by prateeksingh0001** [comment]
I have rebased the branches atop the upstream its well and good now! :) Please review!

**Comment by Udayan12167** [comment]
ack 516fed4, bbd3e56

**Comment by Nosferatul** [comment]
ack bbd3e56

**Comment by Nosferatul** [comment]
@prateeksingh0001 rebase

**Comment by prateeksingh0001** [comment]
I've rebased the branch, please review. @Nosferatul

**Comment by Nosferatul** [comment]
ack e9d226d b5298b5

**Comment by Nosferatul** [comment]
@rultor merge
